### PR TITLE
Make keep_alive_timeout configurable

### PR DESCRIPTION
**Pull request by nestegg**
Allow keep_alive_timeout to be set in persistent:

HTTP.persistent("http...", timeout: 120)

### CODE REVIEW

**Comment by nestegg** [comment]
This CI failure is unrelated to the changes.

File: lib/http/chainable.rb [hunk 1]
```diff
@@ -138,8 +140,11 @@ def timeout(klass, options = {}) # rubocop:disable Style/OptionHash
     #
     #   @yieldparam [HTTP::Client] client Persistent client
     #   @return [Object] result of last expression in the block
-    def persistent(host)
```
Comment (ixti) [inline lib/http/chainable.rb]: We officially support ruby 2.0+, so it's better to use kwargs in here:

``` ruby
def persistent(host, timeout: nil)
  # ...
```

File: spec/lib/http_spec.rb [hunk 2]
```diff
@@ -288,6 +289,14 @@
         end
       end
     end
+
+    context "with timeout specified" do
+      subject { HTTP.persistent host, :timeout => 100 }
+      it "sets keep_alive_timeout" do
+        options = subject.default_options
```
Comment (ixti) [inline spec/lib/http_spec.rb]: use named subject please:

``` ruby
subject(:client) { HTTP.persistent ... }
# ...
options = client.default_options
```

File: spec/lib/http_spec.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# coding: utf-8
```
Comment (ixti) [inline spec/lib/http_spec.rb]: this duplicates line 3

**Comment by nestegg** [comment]
This has been updated.

**Comment by ixti** [comment]
Merged as https://github.com/httprb/http/commit/4738d22c4b615aba193710aee83cc34b5c4792de

**Pull request closed without merging**